### DISCUSSION TITLE
Docstrings

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#40] [Ideas]
Hi, are you planning to add docstrings for methods?
i.e. docstrings with a description of what the method does.

**Comment by CEXT-Dan** [comment]
Yes, I have been thinking about this. I don’t want to do this by hand, except maybe for some obscure functions, there’s just too much. I need to figure out some method of extracting this data directly from Autodesk ObjectARX Reference Guide .CHM
Maybe PyCHM, try to get this when generating stubs

For now, the best way to get help is to download the ObjectARX SDK/documentation
What I do is, I have a shortcut to Autodesk ObjectARX Reference Guide .CHM on my desktop
Example, if I need help with Ed.Core.getCurrentSelectionSet, I search the .chm for  *getCurrentSelectionSet
Which brings up acedGetCurrentSelectionSet

**Comment by CEXT-Dan** [comment]
I used BeautifulSoup to extract the description tag,
I put the contents into a SQLite database. I query the description when I generate the .PYI files
![ds](https://github.com/CEXT-Dan/PyRx/assets/26830697/14a4d6db-25d2-4d90-890d-ec26e957fcbe)

**Reply by gswifort** [reply]
Looks good!
Later, you can also think about parsing the docstring to replace C++ classes with Python classes, e.g. `AcDb::kForRead → PyDb.OpenMode.kForRead`

**Comment by CEXT-Dan** [comment]
The stubs files will grow very large, probably well over 10mb for PyDb.pyi
I hope VS code can handle it. 

I plan on removing those HTML help files, I think they are pretty much useless
I’ll add the docstrings over time, there’s around 10,000.

**Reply by gswifort** [reply]
I don't think this will be a problem for vscode.

Yes, html files seem redundant.
